### ISSUE TITLE
C# glue error (not binding related, macOS deprecation error)

### ISSUE THREAD

**Issue by gitcatrat** [godotengine/godot#37567]
**Godot version:**
Latest, last merge: https://github.com/godotengine/godot/pull/37554

**OS/device including version:**
2017 MacBook Pro, macOS Catalina, 10.15.4

**Issue description:**
macOS related deprecation error:
```
[ 27%] Compiling ==> modules/mono/utils/osx_utils.cpp
modules/mono/utils/osx_utils.cpp:44:20: error: 'LSFindApplicationForInfo' is deprecated: first deprecated in macOS 10.10 - Use LSCopyApplicationURLsForBundleIdentifier instead.
      [-Werror,-Wdeprecated-declarations]
        OSStatus result = LSFindApplicationForInfo(kLSUnknownCreator, bundle_id, nullptr, nullptr, &app_url);
                          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSInfoDeprecated.h:765:1: note: 
      'LSFindApplicationForInfo' has been explicitly marked deprecated here
LSFindApplicationForInfo(
^
1 error generated.
scons: *** [modules/mono/utils/osx_utils.osx.tools.64.o] Error 1
scons: building terminated because of errors.
```

**Steps to reproduce:**
(Probably) get a macOS with version 10.10 or up and run this:
```
scons p=osx tools=yes module_mono_enabled=yes mono_glue=no
```

**Minimal reproduction project:**
Doesn't need one.

**Comment by asmaloney** [comment]
This was fixed by 5d54c015b2dd0a26237c00788be25b3da4cfe3dd.

**Comment by akien-mga** [comment]
Fixed by #40261.
